File: ui/src/app/models/user-tournement.ts
import { Column, Entity, ManyToOne } from 'entitype';

import { Tournement } from './tournement';
import { User } from './user';

@Entity('user_tournement')
export class UserTournement {

  @Column({ columnName: `user_id`, type: `int(11)`, nullable: false, primaryKey: true, default: null })
  _userId: number;

  @Column({ columnName: `tournement_id`, type: `int(11)`, nullable: false, primaryKey: true, default: null })
  _tournementId: number;

  @Column({ columnName: `user_point`, type: `int(11)`, default: null })
  _userPoint?: number;

  @Column({ columnName: `user_victory`, type: `int(11)`, default: null })
  _userVictory?: number;

  @Column({ columnName: `user_null`, type: `int(11)`, default: null })
  _userNull?: number;

  @Column({ columnName: `user_defeat`, type: `int(11)`, default: null })
  _userDefeat?: number;

  @Column({ columnName: `user_goal_scored`, type: `int(11)`, default: null })
  _userGoalScored?: number;

  @Column({ columnName: `user_difference`, type: `int(11)`, default: null })
  _userDifference?: number;

  @Column({ columnName: `user_goal_by_match`, type: `float`, default: null })
  _userGoalByMatch?: number;

  @Column({ columnName: `user_accepted_goal`, type: `int(11)`, default: null })
  _userAcceptedGoal?: number;

  @Column({ columnName: `user_name`, type: `varchar(45)`, default: null })
  _userName?: string;

  @ManyToOne(type => UserTournement, x => x._tournementId)
  _tournement: Tournement;

  @ManyToOne(type => UserTournement, x => x._userId)
  _user: User;


  public get userId(): number {
    return this._userId;
  }

  public set userId(value: number) {
    this._userId = value;
  }

  public get tournementId(): number {
    return this._tournementId;
  }

  public set tournementId(value: number) {
    this._tournementId = value;
  }

  public get userPoint(): number {
    return this._userPoint;
  }

  public set userPoint(value: number) {
    this._userPoint = value;
  }

  public get userVictory(): number {
    return this._userVictory;
  }

  public set userVictory(value: number) {
    this._userVictory = value;
  }

  public get userNull(): number {
    return this._userNull;
  }

  public set userNull(value: number) {
    this._userNull = value;
  }

  public get userDefeat(): number {
    return this._userDefeat;
  }

  public set userDefeat(value: number) {
    this._userDefeat = value;
  }

  public get userGoalScored(): number {
    return this._userGoalScored;
  }

  public set userGoalScored(value: number) {
    this._userGoalScored = value;
  }

  public get userDifference(): number {
    return this._userDifference;
  }

  public set userDifference(value: number) {
    this._userDifference = value;
  }

  public get userGoalByMatch(): number {
    return this._userGoalByMatch;
  }

  public set userGoalByMatch(value: number) {
    this._userGoalByMatch = value;
  }

  public get userAcceptedGoal(): number {
    return this._userAcceptedGoal;
  }

  public set userAcceptedGoal(value: number) {
    this._userAcceptedGoal = value;
  }

  public get userName(): string {
    return this._userName;
  }

  public set userName(value: string) {
    this._userName = value;
  }

  public get tournement(): Tournement {
    return this._tournement;
  }

  public set tournement(value: Tournement) {
    this._tournement = value;
  }

  public get user(): User {
    return this._user;
  }

  public set user(value: User) {
    this._user = value;
  }
}

File: app/com/models/UserTournement.java
package com.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.io.Serializable;
import javax.persistence.*;


/**
 * The persistent class for the user_tournement database table.
 * 
 */
@Entity
@Table(name="user_tournement")
@NamedQuery(name="UserTournement.findAll", query="SELECT u FROM UserTournement u")
public class UserTournement implements Serializable {
	private static final long serialVersionUID = 1L;

	@EmbeddedId
	private UserTournementPK id;

	@Column(name="user_accepted_goal")
	private int userAcceptedGoal;

	@Column(name="user_defeat")
	private int userDefeat;

	@Column(name="user_difference")
	private int userDifference;

	@Column(name="user_goal_by_match")
	private float userGoalByMatch;

	@Column(name="user_goal_scored")
	private int userGoalScored;

	@Column(name="user_name")
	private String userName;

	@Column(name="user_null")
	private int userNull;

	@Column(name="user_point")
	private int userPoint;

	@Column(name="user_victory")
	private int userVictory;

	//bi-directional many-to-one association to Tournement
	@JsonIgnore
	@ManyToOne
	@JoinColumn(name="tournement_id")
	private Tournement tournement;

	//bi-directional many-to-one association to User
	@ManyToOne
	@JoinColumns({
									@JoinColumn(name="user_id", referencedColumnName="user_id"),
									@JoinColumn(name="user_pseudo", referencedColumnName="user_pseudo")
	})
	private User user;

	public UserTournement() {
	}

	public UserTournementPK getId() {
		return this.id;
	}

	public void setId(UserTournementPK id) {
		this.id = id;
	}

	public int getUserAcceptedGoal() {
		return this.userAcceptedGoal;
	}

	public void setUserAcceptedGoal(int userAcceptedGoal) {
		this.userAcceptedGoal = userAcceptedGoal;
	}

	public int getUserDefeat() {
		return this.userDefeat;
	}

	public void setUserDefeat(int userDefeat) {
		this.userDefeat = userDefeat;
	}

	public int getUserDifference() {
		return this.userDifference;
	}

	public void setUserDifference(int userDifference) {
		this.userDifference = userDifference;
	}

	public float getUserGoalByMatch() {
		return this.userGoalByMatch;
	}

	public void setUserGoalByMatch(float userGoalByMatch) {
		this.userGoalByMatch = userGoalByMatch;
	}

	public int getUserGoalScored() {
		return this.userGoalScored;
	}

	public void setUserGoalScored(int userGoalScored) {
		this.userGoalScored = userGoalScored;
	}

	public String getUserName() {
		return this.userName;
	}

	public void setUserName(String userName) {
		this.userName = userName;
	}

	public int getUserNull() {
		return this.userNull;
	}

	public void setUserNull(int userNull) {
		this.userNull = userNull;
	}

	public int getUserPoint() {
		return this.userPoint;
	}

	public void setUserPoint(int userPoint) {
		this.userPoint = userPoint;
	}

	public int getUserVictory() {
		return this.userVictory;
	}

	public void setUserVictory(int userVictory) {
		this.userVictory = userVictory;
	}

	public Tournement getTournement() {
		return this.tournement;
	}

	public void setTournement(Tournement tournement) {
		this.tournement = tournement;
	}

	public User getUser() {
		return this.user;
	}

	public void setUser(User user) {
		this.user = user;
	}

}